Update Contributing Guidelines (#252)
JoshOrndorff/recipes: CONTRIBUTING.md

# Contributing Guidelines

The Substrate Recipes strive to identify and extract useful patterns and examples of developing on Substrate, and to present those patterns in an approachable and fun format. **Your help is Welcome!**

The Recipes are part of the Substrate Developer Hub, so inter-linking between the Recipes and other DevHub facets is common. In particular, the Recipes frequently links to:
* The [Reference Docs](https://substrate.dev/rustdocs/master/)
* The [Tutorials](https://substrate.io/tutorials/)

> * The [Knowledge Base](https://substrate.dev/docs)

Reviewer: ```suggestion
* The [Knowledge Base](https://substrate.io/kb/learn-substrate)
```

### Master Branch

The `master` branch contains stable, published code and is where release versions are tagged. Released versions will always be in the history of the `master` branch. The code on `master` uses published dependencies from `crates.io` and uses git dependencies only where absolutely necessary because relevant crates are not published on crates.

### Develop Branch
The `develop` branch is where active development happens. It is where new recipes, restructures, revisions, CI updates, and most other changes are merged. In order to keep up with the latest Substrate development, the `develop` branch allows dependencies from git.

### Cutting a Release

It is time  to tag a new release when either, enough new features have been contributed that releasing makes sense, or, more likely, Substrate itself has tagged a new release. The release process also follows gitflow. The process of creating a new release is usually undertaken by the project maintainer, but the steps are outlined here nonetheless.

1. Create a release branch off of `develop`.
2. Update dependencies to crates.io.
3. Update version of each Recipes crate.

> 4. Update version listed in the main `README.md`.
> 5. Make a pull request against `master`.
> 6. Tag the new release version in the history of `master` only after the release branch is merged.

Author: ```suggestion
4. Make a pull request against `master`.
5. Tag the new release version in the history of `master` only after the release branch is merged.
```

* **New recipes** - If you know how to do something useful in Substrate that is not yet covered in the Recipes, please contribute.
* **UX improvements** - Any way to make it easier and less confusing to get new users on boarded is welcome.
* **CI Improvements** - The more tests we have automated, the higher quality the Recipes will be.

## Style

### Rust Code
There is not yet strict enforcement of the [Rust in Substrate coding style](https://wiki.parity.io/Substrate-Style-Guide), but it is highly encouraged to wrap lines at 120 characters a line (or less) for improving reviewer experience on github.

Graciously invoke `cargo fmt` and `cargo clippy` on any Rust code; This should soon be enforced by CI.

### Cargo.toml
Prefer listing dependencies under a single `[dependencies]` header in lieu of using a `[dependencies.some_import]` for every `some_import` module imported.

### English
No standards for language style are enforced aside from the common English spelling/grammar rules. @4meta5 has a few *preferences*:
* Avoid using "we", "our", "you" because it often is conducive to unnecessary language
* Prefer active voice ("you may want to use active voice" `=>` "use active voice")

> * Link as often as possible to outside content and useful resources including other recipes, knowledgebase,

Reviewer: ```suggestion
* Link as often as possible to outside content and useful resources including other recipes, knowledge base,
```
